agent/src/agent/pipeline/json_builder/json_builder.py

```diff
@@ -127,27 +133,67 @@ def _validate_config_for_create(config: dict):
     json_schema = {
         'type': 'object',
         'properties': {
-            'source': {'type': 'string', 'enum': source.repository.get_all_names()},
-            'pipeline_id': {'type': 'string', 'minLength': 1, 'maxLength': 100}
+            'source': {
+                'type': 'string',
+                'enum': source.repository.get_all_names()
+            },
+            'pipeline_id': {
+                'type': 'string',
+                'minLength': 1,
+                'maxLength': 100
+            }
         },
         'required': ['source', 'pipeline_id']
     }
     jsonschema.validate(config, json_schema)
 
 
 def _load_config(pipeline_: Pipeline, config: dict, is_edit=False):
-    if isinstance(pipeline_, pipeline.RawPipeline):
-        config['uses_schema'] = False
-    elif 'uses_schema' not in config:
-        if is_edit:
-            config['uses_schema'] = pipeline_.config.get('uses_schema', False)
-        else:
-            config['uses_schema'] = pipeline.manager.supports_schema(pipeline_)
+    config['uses_schema'] = _uses_schema(pipeline_, config, is_edit)
     json_builder.get(pipeline_, config, is_edit).build()
     # todo too many validations, 4 validations here
     pipeline.config.validators.get_config_validator(pipeline_.source.type).validate(pipeline_)
 
 
+class _SchemaChooser:
+    @staticmethod
+    def choose(pipeline_: Pipeline, config: dict, is_edit=False) -> bool:
+        if isinstance(pipeline_, pipeline.RawPipeline):
```

Comment:
shouldn't we add the same condition here for TestPipeline also? if we decide that test pipelines don't use schema

Reply:
it was there -_-, probably I accidentally overwrote it when force-pushed

Reply:
yes, it's also like a duplicate code, I'll try to avoid it